conn2res/iodata.py

```diff
@@ -190,67 +204,122 @@ def split_dataset(data, frac_train=0.7):
         array with test observations
     """
 
-    n_train = int(frac_train * len(data))
+    argout = []
+    for arg in args:
+        if isinstance(arg, list):
+            n_train = int(frac_train * len(arg))
+
+            if axis == 0:
+                argout.extend([np.vstack(arg[:n_train]),
+                              np.vstack(arg[n_train:])])
+            elif axis == 1:
+                argout.extend([np.hstack(arg[:n_train]),
+                              np.hstack(arg[n_train:])])
+
+        elif isinstance(arg, np.ndarray):
+            n_train = int(frac_train * arg.shape[axis])
+
+            if axis == 0:
+                argout.extend([arg[:n_train, :], arg[n_train:, :]])
+            elif axis == 1:
+                argout.extend([arg[:, :n_train], arg[:, n_train:]])
 
-    if isinstance(data, list):
-        return np.vstack(data[:n_train]), np.vstack(data[n_train:])
-    else:
-        return data[:n_train], data[n_train:]
+    return tuple(argout)
 
 
 def visualize_data(task, x, y, plot=True):
     """
-    #TODO
+    # TODO
     Visualizes dataset for task
 
     Parameters
     ----------
     task : str
     x,y  : list or numpy.ndarray
-        
+
     """
 
     print(f'\n----- {task.upper()} -----')
     print(f'\tNumber of trials = {len(x)}')
 
     # convert x and y to arrays for visualization
-    if isinstance(x, list): x = np.vstack(x[:5])
-    if isinstance(y, list): y = np.vstack(y[:5]).squeeze()
+    if isinstance(x, list):
+        x = np.vstack(x[:5])
+    if isinstance(y, list):
+        y = np.vstack(y[:5]).squeeze()
 
     print(f'\tinputs shape = {x.shape}')
     print(f'\tlabels shape = {y.shape}')
 
-    # number of features, labels and classes 
-    try: n_features = x.shape[1]
-    except: n_features = 1
+    # number of features, labels and classes
+    try:
+        n_features = x.shape[1]
+    except:
+        n_features = 1
 
-    try: n_labels = y.shape[1]
-    except: n_labels = 1
+    try:
+        n_labels = y.shape[1]
+    except:
+        n_labels = 1
     n_classes = len(np.unique(y))
 
     model = select_model(y)
-    
+
     print(f'\t  n_features = {n_features}')
     print(f'\tn_labels   = {n_labels}')
     print(f'\tn_classes  = {n_classes}')
     print(f'\tlabel type : {y.dtype}')
     print(f'\tmodel = {model.__name__}')
 
     if plot:
-        fig = plt.figure(num=1, figsize=(12,10))
-        ax = plt.subplot(111)
-
-        x_labels  = [f'I{n+1}' for n in range(n_features)]
-        y_labels = [f'O{n+1}' for n in range(n_labels)]
-
-        plt.plot(x[:], label=x_labels)
-        plt.plot(y[:], label=y_labels)
-        plt.legend()
-        plt.suptitle(task)
-        
-        sns.despine(offset=10, trim=True)
-        # fig.savefig(fname=f'figs/{task}_io.png', transparent=True, bbox_inches='tight', dpi=300)
-        plt.show()
-        plt.close()
-    
+        plotting.plot_task(x, y, task)
+
+
+def get_sample_weight(inputs, labels, sample_block=None):

```

Comment:
Sorry, the docstring of the function is outdated. This function finds blocks in the data based on repetition in the signal then it defines a sample_weight proportional to the length of this block. For instance, the signal [1 1 1 1 2] would produce sample_weight [0.25 0.25 0.25 0.25 1].